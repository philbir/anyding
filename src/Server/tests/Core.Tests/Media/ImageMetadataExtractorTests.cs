using Anyding.Media;
using Snapshooter.Xunit;

namespace Media.Tests;

public class ImageMetadataExtractorTests
{
    [Fact]
    public async Task GetMetadata()
    {
        // Arrange
        var extractor = new ImageMetadataExtractor();

        // Act
        ImageMetadata metadata = await extractor.GetMetadataAsync(
            TestData.Images.JpgIphone15,
            CancellationToken.None);

        // Assert
        metadata.MatchSnapshot();
    }
}

