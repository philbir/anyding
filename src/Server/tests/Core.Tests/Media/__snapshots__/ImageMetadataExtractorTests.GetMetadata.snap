{
  "ImageId": null,
  "Dimension": {
    "Height": 4284,
    "Width": 5712
  },
  "GeoLocation": {
    "Latitude": 47.034722,
    "Longitude": 9.006111,
    "Altitude": 848,
    "Source": null
  },
  "Orientation": "Horizontal (normal)",
  "Camera": {
    "Model": "iPhone 15 Pro Max",
    "Make": "Apple",
    "Software": "18.1"
  },
  "DateTaken": "2024-09-06T09:14:09",
  "Lens": "iPhone 15 Pro Max back triple camera 6.765mm f/1.78"
}
